src/libcore/marker.rs

```diff
@@ -672,6 +673,37 @@ mod impls {
     unsafe impl<T: Send + ?Sized> Send for &mut T {}
 }
 
+/// Compiler-internal trait used to indicate the type of enum discriminants.
+///
+/// This trait is automatically implemented for every type and does not add any
+/// guarantees to [`mem::Discriminant`]. It is **undefined behavior** to transmute
+/// between `DiscriminantKind::Discriminant` and `mem::Discriminant`.
+///
+/// [`mem::Discriminant`]: https://doc.rust-lang.org/stable/core/mem/struct.Discriminant.html
+#[unstable(
+    feature = "discriminant_kind",
+    issue = "none",
+    reason = "this trait is unlikely to ever be stabilized, use `mem::discriminant` instead"
```

Comment:
```suggestion
    reason = "this trait is unlikely to ever be stabilized, use `mem::Discriminant` instead"
```

Reply:
I intentionally used `mem::discriminant` here, as I meant the method, not the type. Don't care either way though :shrug:

Reply:
I thought you meant the type, as this trait is used to get the type of the discriminant.

Reply:
hmm. This message is not optimal :thinking: 

Would something like ```this trait is unlikely to ever be stabilized, try using `mem::discriminant` directly``` be better?